examples/compiled/cat.sh

```diff
@@ -195,7 +181,7 @@ _open() { # $2: filename, $3: flags, $4: mode
     : $((__cursor_fd$__fd = 0))         # Make buffer empty
     : $((__buflen_fd$__fd = 1000))      # Init buffer length
     : $((__state_fd$__fd = $3))         # Mark the fd as opened
-    pack_string $2
+    __res=$(_put_pstr __ $2)
```

Comment:
This may be slower on certain shells, but opening a file is a relatively rare operation so it feels like a good compromise.